### PR TITLE
Tigerbrew

### PR DESCRIPTION
**Pull request by mistydemeo**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully ran `brew tests` with your changes locally?

These patches originated in Tigerbrew; they add support for obsolete Xcode and XQuartz versions. My goal is to eventually unify the Homebrew and Tigerbrew core code, so I'd like to get support for this into Homebrew.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
All these changes seem non-invasive and reasonable. They may even end up helping Linuxbrew too 👍

File: Library/Homebrew/os/mac/xquartz.rb [hunk 1]
```diff
@@ -98,6 +99,9 @@ def prefix
           Pathname.new("/opt/X11")
         elsif Pathname.new("/usr/X11/lib/libpng.dylib").exist?
           Pathname.new("/usr/X11")
+        # X11 doesn't include libpng on Tiger
+        elsif File.exist?("/usr/X11R6/lib/libX11.dylib")
```
Comment (xu-cheng) [inline Library/Homebrew/os/mac/xquartz.rb]: Style nit. Probably want to use same `Pathname#exist?` or vice verse to keep the consistence.

**Comment by xu-cheng** [comment]
:+1:

File: Library/Homebrew/os/mac/xquartz.rb (hunk 1 above)
Comment (mistydemeo) [inline Library/Homebrew/os/mac/xquartz.rb]: Updated!

**Comment by apjanke** [comment]
:+1:

**Pull request closed without merging**